test/performance/alignment/edit_distance_unbanded_benchmark.cpp

```diff
@@ -61,15 +66,21 @@ void seqan3_edit_distance_dna4(benchmark::State & state)
     auto seq2 = generate_sequence<seqan3::dna4>(sequence_length, 0, 1);
     int score = 0;
 
-    using edit_traits_t = seqan3::detail::default_edit_distance_trait_type<std::add_lvalue_reference_t<decltype(seq1)>,
-                                                                           std::add_lvalue_reference_t<decltype(seq2)>,
+    using seq1_ref_t = std::add_lvalue_reference_t<decltype(seq1)>;
+    using seq2_ref_t = std::add_lvalue_reference_t<decltype(seq2)>;
+    using alignment_result_t = alignment_result_type_t<seq1_ref_t, seq2_ref_t, decltype(edit_distance_cfg)>;
+
+    using edit_traits_t = seqan3::detail::default_edit_distance_trait_type<seq1_ref_t,
+                                                                           seq2_ref_t,
                                                                            decltype(edit_distance_cfg),
+                                                                           alignment_result_t,
                                                                            std::false_type>;
 
+
     for (auto _ : state)
     {
         seqan3::detail::edit_distance_unbanded edit_distance{seq1, seq2, edit_distance_cfg, edit_traits_t{}};
-        edit_distance(0u);
+        edit_distance(0u, [] (auto) {/*noop*/});
```

Comment:
I think the "do nothing", as you used in `test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp` is a bit nicer than "noop"